pkg/provider/naming.go

```diff
@@ -0,0 +1,44 @@
+package provider
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/pulumi/pulumi/pkg/tokens"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var renameBlacklist = sets.NewString("v1/Namespace")
+
+// assignName generates a name for an object if it doesn't already have one. Uses DNS-1123-compliant
+// characters.
+func assignName(obj *unstructured.Unstructured, base tokens.QName) {
+	if !renameBlacklist.Has(obj.GroupVersionKind().String()) {
```

Comment:
Okay, so for my own understanding, this reads to me as "autogenerate a random suffix for this thing's name if it's not a v1/Namespace". I thought [here](https://github.com/pulumi/pulumi-kubernetes/issues/10#issuecomment-407160010) you said that `v1/Namespace` isn't an exception - should we be doing the same for those, too?

Reply:
I haven't updated this yet, I'm removing it now.